scripts/cpu_temp.sh

```diff
@@ -18,7 +18,16 @@ print_cpu_temp() {
     else
       val="$(sensors)"
     fi
-    echo "$val" | sed -e 's/^Tccd/Core /' | awk -v format="$cpu_temp_format$cpu_temp_unit" '/^Core [0-9]+/ {gsub("[^0-9.]", "", $3); sum+=$3; n+=1} END {printf(format, sum/n)}'
+    val=$(echo "$val" | sed -e 's/^Tccd/Core /')
+    if [[ $val =~ Core ]]; then
+      echo "$val" | awk -v format="$cpu_temp_format$cpu_temp_unit" '/^Core [0-9]+/ {gsub("[^0-9.]", "", $3); sum+=$3; n+=1} END {printf(format, sum/n)}'
+    elif [[ $val =~ Tctl ]]; then
+      echo "$val" | sed -n 's/Tctl:\s*+//p' | tr -d " "
+    elif [[ $val =~ Tdie ]]; then
+      echo "$val" | sed -n 's/Tdie:\s*+//p' | tr -d " "
+    else
+      echo "NA"
+    fi
```

Comment:
maybe this?

```suggestion
    echo "$val" | sed -e 's/^Tccd/Core /' -e 's/^Tctl:/Core /' -e 's/^Tdie:/Core /' | awk -v format="$cpu_temp_format$cpu_temp_unit" '/^Core\s+[0-9]+/ {gsub("[^0-9.]", "", $3); sum+=$3; n+=1} END {printf(format, sum/n)}'
```

Reply:
This is not working for my CPU (AMD Ryzen 9 3950X). The original proposal somehow works for me.

Reply:
Hmm. Can you post the output of `sensors`?

Reply:
Sorry, I'm a little bit busy lately. Here's my `sensors` output:
```
➜ sensors
iwlwifi_1-virtual-0
Adapter: Virtual device
temp1:            N/A

k10temp-pci-00c3
Adapter: PCI adapter
Tdie:         +57.8°C  (high = +70.0°C)
Tctl:         +57.8°C
```